pkgs/by-name/ex/extract_otp_secrets/package.nix

```
@@ -0,0 +1,89 @@
+{
+  pkgs,
+  lib,
+  copyDesktopItems,
+  fetchFromGitHub,
+  makeDesktopItem,
+  python3Packages,
+}:
+
+let
+  pyPkgs = python3Packages.overrideScope (
+    self: super: {
+      opencv4 = super.opencv4.override {
+        enableGtk3 = true;
+      };
+    }
+  );
+in
+pyPkgs.buildPythonApplication rec {
+  pname = "extract_otp_secrets";
+  version = "2.10.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "scito";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-zh3SVHIEDsl4iF8cuvY42XElhcso34DxpQWtTZB/uBI=";
+  };
+
+  build-system = with pyPkgs; [
+    pytestCheckHook
+    pythonRelaxDepsHook
+    setuptools
+    setuptools-git-versioning
+  ];
+
+  dependencies = with pyPkgs; [
+    colorama
+    opencv4
+    pillow
+    protobuf
+    pyzbar
+    qrcode
+    qreader_1
+  ];
+
+  pythonRemoveDeps = [
+    "opencv-contrib-python"
+  ];
+
+  patches = [
+    # Dependencies used only for the binary package
+    ./dependencies.diff
+    # https://github.com/scito/extract_otp_secrets/pull/444
+    ./detect_close_window.diff
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+
+  nativeCheckInputs = with pyPkgs; [
+    pytest-mock
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      desktopName = pname;
+      name = pname;
+      exec = pname;
+    })
+  ];
+
+  meta = with lib; {
+    description = "Extracts OTP secrets from QR codes exported by 2FA apps";
+    longDescription = ''
+      Extracts one time password (OTP) secrets from QR codes exported by
+      two-factor authentication (2FA) apps such as "Google Authenticator". The
+      exported QR codes from authentication apps can be captured by camera,
+      read from images, or read from text files. The secrets can be exported to
+      JSON or CSV, or printed as QR codes to console.
+    '';
+    license = licenses.gpl3;
+    homepage = "https://github.com/scito/extract_otp_secrets";
+    maintainers = with maintainers; [ nova ];
+    mainProgram = pname;
+  };
```

```suggestion
  meta = {
    description = "Extracts OTP secrets from QR codes exported by 2FA apps";
    longDescription = ''
      Extracts one time password (OTP) secrets from QR codes exported by
      two-factor authentication (2FA) apps such as "Google Authenticator". The
      exported QR codes from authentication apps can be captured by camera,
      read from images, or read from text files. The secrets can be exported to
      JSON or CSV, or printed as QR codes to console.
    '';
    license = lib.licenses.gpl3Plus;
    homepage = "https://github.com/scito/extract_otp_secrets";
    maintainers = with lib.maintainers; [ nova ];
    mainProgram = pname;
  };
```
`lib.licenses.gplN` (where N is 2 or 3) are deprecated now, specify `gplNOnly` or `gplNPlus`. Judging by [the source](https://github.com/scito/extract_otp_secrets/blob/4d0b5db83f092fbc5267675c4b1a2b038350c3aa/src/extract_otp_secrets.py#L23), it's the latter